src/main/java/com/labsynch/labseer/service/BulkLoadServiceImpl.java

```diff
@@ -2051,84 +2055,107 @@ public String generateSummaryText(int numRecordsRead,
-		numberOfParents = parents.size();
-		String manuallyRegistered = "Manually Registered";
-		for (Parent parent : parents) {
-			acasDependencies.put(parent.getCorpName(), new HashSet<String>());
-			if (parent.getSaltForms() != null) {
-				for (SaltForm saltForm : parent.getSaltForms()) {
-					acasDependencies.put(saltForm.getCorpName(), new HashSet<String>());
-					if (saltForm.getBulkLoadFile() != null && saltForm.getBulkLoadFile() != bulkLoadFile) {
-						if (cmpdRegDependencies.containsKey(parent.getCorpName())) {
-							cmpdRegDependencies.get(parent.getCorpName()).add(saltForm.getBulkLoadFile().getFileName());
-						} else {
-							HashSet<String> dependentFiles = new HashSet<String>();
-							dependentFiles.add(saltForm.getBulkLoadFile().getFileName());
-							cmpdRegDependencies.put(parent.getCorpName(), dependentFiles);
-						}
-					}
-					if (saltForm.getLots() != null) {
-						for (Lot lot : saltForm.getLots()) {
-							acasDependencies.put(lot.getCorpName(), new HashSet<String>()); 
-							if (lot.getBulkLoadFile() == null) {
-								dependentSingleRegLots.add(lot.getCorpName());
-							} else if (lot.getBulkLoadFile() != bulkLoadFile) {
-								if (cmpdRegDependencies.containsKey(parent.getCorpName())) {
-									cmpdRegDependencies.get(parent.getCorpName())
-											.add(lot.getBulkLoadFile().getFileName());
-								} else {
-									HashSet<String> dependentFiles = new HashSet<String>();
-									dependentFiles.add(lot.getBulkLoadFile().getFileName());
-									cmpdRegDependencies.put(parent.getCorpName(), dependentFiles);
-								}
-							}
-						}
-					}
+		// These Sets are used to count the number of parents, saltForms, and lots directly linked to this bulkLoadFile
+		Set<String> uniqueParentCorpNames = new HashSet<String>();
+		Set<String> uniqueSaltFormCorpNames = new HashSet<String>();
+		Set<String> uniqueLotCorpNames = new HashSet<String>();
+		
+		// Grab the id of the current bulkLoadFile being checked, since we'll reference it a lot below
+		Long bulkLoadFileId = bulkLoadFile.getId();
+		// collect all parent, saltform, and lot corpNames and bulkLoadFileIds tied to this file (directly or indirectly)
+		Collection<BulkLoadParentSaltFormLotDTO> dtos = getBulkLoadDTOsByBulkLoadFileId(bulkLoadFileId);
+		// collect all the unique bulkLoadFile ids, and later build a map of id to name
+		Map<Long, String> bulkLoadIdsToNames = new HashMap<Long, String>();
+		
+		for (BulkLoadParentSaltFormLotDTO dto : dtos) {
+			String parentCorpName = dto.getParentCorpName();
+			String saltFormCorpName = dto.getSaltFormCorpName();
+			String lotCorpName = dto.getLotCorpName();
+			Long parentBulkLoadFileId = dto.getParentBulkLoadFileId();
+			Long saltFormBulkLoadFileId = dto.getSaltFormBulkLoadFileId();
+			Long lotBulkLoadFileId = dto.getLotBulkLoadFileId();
+			// acasDependencies, unique corpName lists: collect all parent, saltform, and lot corpnames directly tied to the file
```

Comment:
I've tried to more clearly document what each of these Maps are supposed to contain, so the logic below is easier to follow.